src/world_model/world_model.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+from numpy import random
+from numpy.random import choice
+from gymnasium import Env, ResetNeeded
+
+# TODO: add typing
+
+class WorldModel(Env):
+    """An abstract base class for potentially probabilistic world models, 
+    extending gymnasion.Env by providing methods for enquiring transition probabilities.
+    Most implementations will probably combine this with a gymnasium.Env.
```

Comment:
Thing to consider regarding the class hierarchy.

If it is the case that "most implementations" will combine this with a gym.Env, then WorldModel shouldn't inherit from Env.

If we want it to be required that every WorldModel "combines" this with gym.Env, then the current inheritance might make sense